### PR TITLE
Use Checks.checkNotNull() in all Query Builders

### LINKED ISSUE #181: Use Checks.checkNotNull() in all Query Builders

### PR DESCRIPTION
**Pull request by nikitin-da**
Closes #181 
@artem-zinnatullin PTAL

### CODE REVIEW

File: storio-common/src/main/java/com/pushtorefresh/storio/util/QueryUtil.java [hunk 1]
```diff
@@ -21,14 +21,14 @@ private QueryUtil() {
...
      */
     @Nullable
-    public static List<String> varargsToList(@Nullable Object... args) {
+    public static List<String> varargsToList(@Nullable Object[] args) {
         if (args == null || args.length == 0) {
             return null;
         } else {
             final List<String> list = new ArrayList<>(args.length);
 
-            for (int i = 0; i < args.length; i++) {
```
Comment (artem-zinnatullin) [inline storio-common/src/main/java/com/pushtorefresh/storio/util/QueryUtil.java]: This code will be faster on Android, unfortunately, Java compiler won't optimise this :(

File: storio-contentprovider/src/main/java/com/pushtorefresh/storio/contentprovider/operation/get/PreparedGetCursor.java [hunk 2]
```diff
@@ -102,9 +103,7 @@ public Builder withGetResolver(@Nullable GetResolver getResolver) {
          */
         @NonNull
         public PreparedOperationWithReactiveStream<Cursor> prepare() {
-            if (query == null) {
-                throw new IllegalStateException("Please specify query");
-            }
+            Checks.checkNotNull(query, "Please specify query");
```
Comment (artem-zinnatullin) [inline storio-contentprovider/src/main/java/com/pushtorefresh/storio/contentprovider/operation/get/PreparedGetCursor.java]: Static method import please!

File: storio-contentprovider/src/main/java/com/pushtorefresh/storio/contentprovider/operation/get/PreparedGetListOfObjects.java [hunk 3]
```diff
@@ -176,13 +177,8 @@ public Builder(@NonNull StorIOContentProvider storIOContentProvider, @NonNull Cl
...
         @NonNull
         public PreparedOperationWithReactiveStream<List<T>> prepare() {
-            if (mapFunc == null) {
-                throw new IllegalStateException("Please specify map function");
-            }
-
-            if (query == null) {
-                throw new IllegalStateException("Please specify query");
-            }
+            Checks.checkNotNull(mapFunc, "Please specify map function");
```
Comment (artem-zinnatullin) [inline storio-contentprovider/src/main/java/com/pushtorefresh/storio/contentprovider/operation/get/PreparedGetListOfObjects.java]: Static import

File: storio-contentprovider/src/main/java/com/pushtorefresh/storio/contentprovider/operation/get/PreparedGetListOfObjects.java [hunk 4]
```diff
@@ -176,13 +177,8 @@ public Builder(@NonNull StorIOContentProvider storIOContentProvider, @NonNull Cl
...
+            Checks.checkNotNull(query, "Please specify query");
```
Comment (artem-zinnatullin) [inline storio-contentprovider/src/main/java/com/pushtorefresh/storio/contentprovider/operation/get/PreparedGetListOfObjects.java]: Static import

File: storio-contentprovider/src/main/java/com/pushtorefresh/storio/contentprovider/query/DeleteQuery.java [hunk 5]
```diff
@@ -129,9 +130,7 @@ public Builder whereArgs(@Nullable Object... whereArgs) {
          */
         @NonNull
         public DeleteQuery build() {
-            if (uri == null) {
-                throw new IllegalStateException("Please specify uri");
-            }
+            Checks.checkNotNull(uri, "Please specify uri");
```
Comment (artem-zinnatullin) [inline storio-contentprovider/src/main/java/com/pushtorefresh/storio/contentprovider/query/DeleteQuery.java]: Static import

File: storio-contentprovider/src/main/java/com/pushtorefresh/storio/contentprovider/query/InsertQuery.java [hunk 6]
```diff
@@ -75,9 +77,7 @@ public Builder uri(@NonNull Uri uri) {
          */
         @NonNull
         public InsertQuery build() {
-            if (uri == null) {
-                throw new IllegalStateException("Please specify uri");
-            }
+            Checks.checkNotNull(uri, "Please specify uri");
```
Comment (artem-zinnatullin) [inline storio-contentprovider/src/main/java/com/pushtorefresh/storio/contentprovider/query/InsertQuery.java]: Static import

File: storio-contentprovider/src/main/java/com/pushtorefresh/storio/contentprovider/query/Query.java [hunk 7]
```diff
@@ -201,9 +202,7 @@ public Builder sortOrder(@Nullable String sortOrder) {
          */
         @NonNull
         public Query build() {
-            if (uri == null) {
-                throw new IllegalStateException("Please specify uri");
-            }
+            Checks.checkNotNull(uri, "Please specify uri");
```
Comment (artem-zinnatullin) [inline storio-contentprovider/src/main/java/com/pushtorefresh/storio/contentprovider/query/Query.java]: Static import

File: storio-contentprovider/src/main/java/com/pushtorefresh/storio/contentprovider/query/UpdateQuery.java [hunk 8]
```diff
@@ -120,9 +121,7 @@ public Builder whereArgs(@Nullable Object... whereArgs) {
          */
         @NonNull
         public UpdateQuery build() {
-            if (uri == null) {
-                throw new IllegalStateException("Please specify uri");
-            }
+            Checks.checkNotNull(uri, "Please specify uri");
```
Comment (artem-zinnatullin) [inline storio-contentprovider/src/main/java/com/pushtorefresh/storio/contentprovider/query/UpdateQuery.java]: Static import

File: storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/impl/StorIOSQLiteDbImpl.java [hunk 9]
```diff
@@ -197,9 +198,7 @@ public Builder sqliteOpenHelper(@NonNull SQLiteOpenHelper sqliteOpenHelper) {
 
         @NonNull
         public StorIOSQLiteDbImpl build() {
-            if (db == null) {
-                throw new IllegalStateException("Please specify SQLiteDatabase instance");
-            }
+            Checks.checkNotNull(db, "Please specify SQLiteDatabase instance");
```
Comment (artem-zinnatullin) [inline storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/impl/StorIOSQLiteDbImpl.java]: Static import

File: storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/operation/delete/PreparedDeleteCollectionOfObjects.java [hunk 10]
```diff
@@ -171,9 +172,7 @@ public void call(Subscriber<? super DeleteCollectionOfObjectsResult<T>> subscrib
                 deleteResolver = DefaultDeleteResolver.INSTANCE;
             }
 
-            if (mapFunc == null) {
-                throw new IllegalStateException("Please specify map function");
-            }
+            Checks.checkNotNull(mapFunc, "Please specify map function");
```
Comment (artem-zinnatullin) [inline storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/operation/delete/PreparedDeleteCollectionOfObjects.java]: Static import

File: storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/operation/delete/PreparedDeleteObject.java [hunk 11]
```diff
@@ -97,9 +98,7 @@
                 deleteResolver = DefaultDeleteResolver.INSTANCE;
             }
 
-            if (mapFunc == null) {
-                throw new IllegalStateException("Please specify map function");
-            }
+            Checks.checkNotNull(mapFunc, "Please specify map function");
```
Comment (artem-zinnatullin) [inline storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/operation/delete/PreparedDeleteObject.java]: Static import

File: storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/operation/exec_sql/PreparedExecSql.java [hunk 12]
```diff
@@ -66,9 +67,7 @@ public Builder(@NonNull StorIOSQLiteDb storIOSQLiteDb) {
          * @return {@link PreparedExecSql} instance
          */
         @NonNull public PreparedExecSql prepare() {
-            if (rawQuery == null) {
-                throw new IllegalStateException("Please set query object");
-            }
+            Checks.checkNotNull(rawQuery, "Please set query object");
```
Comment (artem-zinnatullin) [inline storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/operation/exec_sql/PreparedExecSql.java]: Static import

File: storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/operation/get/PreparedGetListOfObjects.java [hunk 13]
```diff
@@ -208,9 +209,7 @@ public void call(Subscriber<? super List<T>> subscriber) {
                 getResolver = DefaultGetResolver.INSTANCE;
             }
 
-            if (mapFunc == null) {
-                throw new IllegalStateException("Please specify map function");
-            }
+            Checks.checkNotNull(mapFunc, "Please specify map function");
```
Comment (artem-zinnatullin) [inline storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/operation/get/PreparedGetListOfObjects.java]: Static import

File: storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/operation/put/PreparedPutContentValues.java [hunk 14]
```diff
@@ -74,9 +75,7 @@
          * @return {@link PreparedPutContentValues} instance
          */
         @NonNull public PreparedPutContentValues prepare() {
-            if (putResolver == null) {
-                throw new IllegalStateException("Please specify put resolver");
-            }
+            Checks.checkNotNull(putResolver, "Please specify put resolver");
```
Comment (artem-zinnatullin) [inline storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/operation/put/PreparedPutContentValues.java]: Static import

File: storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/operation/put/PreparedPutIterableContentValues.java [hunk 15]
```diff
@@ -160,9 +161,7 @@ public Builder dontUseTransaction() {
          */
         @NonNull
         public PreparedPutIterableContentValues prepare() {
-            if (putResolver == null) {
-                throw new IllegalStateException("Please specify put resolver");
-            }
+            Checks.checkNotNull(putResolver, "Please specify put resolver");
```
Comment (artem-zinnatullin) [inline storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/operation/put/PreparedPutIterableContentValues.java]: Static import

File: storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/operation/put/PreparedPutObject.java [hunk 16]
```diff
@@ -100,9 +101,7 @@ public void call(Subscriber<? super PutResult> subscriber) {
          */
         @NonNull
         public PreparedOperation<PutResult> prepare() {
-            if (mapFunc == null) {
-                throw new IllegalStateException("Please specify map function");
-            }
+            Checks.checkNotNull(mapFunc, "Please specify map function");
```
Comment (artem-zinnatullin) [inline storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/operation/put/PreparedPutObject.java]: Static import

File: storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/operation/put/PreparedPutObjects.java [hunk 17]
```diff
@@ -163,13 +164,8 @@ public void call(Subscriber<? super PutCollectionResult<T>> subscriber) {
...
          */
         @NonNull public PreparedPutObjects<T> prepare() {
-            if (mapFunc == null) {
-                throw new IllegalStateException("Please specify map function");
-            }
-
-            if (putResolver == null) {
-                throw new IllegalStateException("Please specify put resolver");
-            }
+            Checks.checkNotNull(mapFunc, "Please specify map function");
```
Comment (artem-zinnatullin) [inline storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/operation/put/PreparedPutObjects.java]: Static import

File: storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/operation/put/PreparedPutObjects.java [hunk 18]
```diff
@@ -163,13 +164,8 @@ public void call(Subscriber<? super PutCollectionResult<T>> subscriber) {
...
+            Checks.checkNotNull(putResolver, "Please specify put resolver");
```
Comment (artem-zinnatullin) [inline storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/operation/put/PreparedPutObjects.java]: Static import

File: storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/query/DeleteQuery.java [hunk 19]
```diff
@@ -141,9 +142,7 @@ public Builder whereArgs(@Nullable Object... whereArgs) {
          */
         @NonNull
         public DeleteQuery build() {
-            if (table == null || table.length() == 0) {
-                throw new IllegalStateException("Please specify table name");
-            }
+            Checks.checkNotEmpty(table, "Please specify table name");
```
Comment (artem-zinnatullin) [inline storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/query/DeleteQuery.java]: Static import

File: storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/query/InsertQuery.java [hunk 20]
```diff
@@ -108,9 +110,7 @@ public Builder nullColumnHack(@Nullable String nullColumnHack) {
          */
         @NonNull
         public InsertQuery build() {
-            if (table == null || table.length() == 0) {
-                throw new IllegalStateException("Please specify table name");
-            }
+            Checks.checkNotEmpty(table, "Please specify table name");
```
Comment (artem-zinnatullin) [inline storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/query/InsertQuery.java]: Static import

File: storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/query/Query.java [hunk 21]
```diff
@@ -340,9 +341,7 @@ public Builder limit(@Nullable String limit) {
          */
         @NonNull
         public Query build() {
-            if (table == null || table.length() == 0) {
-                throw new IllegalStateException("Please specify table name");
-            }
+            Checks.checkNotEmpty(table, "Please specify table name");
```
Comment (artem-zinnatullin) [inline storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/query/Query.java]: Static import

File: storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/query/RawQuery.java [hunk 22]
```diff
@@ -141,9 +142,7 @@ public Builder affectedTables(@NonNull String... tables) {
          */
         @NonNull
         public RawQuery build() {
-            if (query == null || query.length() == 0) {
-                throw new IllegalStateException("Please specify query string");
-            }
+            Checks.checkNotEmpty(query, "Please specify query string");
```
Comment (artem-zinnatullin) [inline storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/query/RawQuery.java]: Static import

File: storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/query/UpdateQuery.java [hunk 23]
```diff
@@ -141,9 +142,7 @@ public Builder whereArgs(@Nullable String... whereArgs) {
          */
         @NonNull
         public UpdateQuery build() {
-            if (table == null || table.length() == 0) {
-                throw new IllegalStateException("Please specify table name");
-            }
+            Checks.checkNotEmpty(table, "Please specify table name");
```
Comment (artem-zinnatullin) [inline storio-sqlitedb/src/main/java/com/pushtorefresh/storio/sqlitedb/query/UpdateQuery.java]: Static import

**Comment by artem-zinnatullin** [comment]
@nikitin-da trolled! :trolleybus:  

Actually, I commented all places with `Static import` to check that all of my comments will be fixed by next PR update and GitHub will hide them :)

**Comment by nikitin-da** [comment]
Check it out

**Comment by artem-zinnatullin** [comment]
Awesome :+1:

**Pull request merged**